server/src/main/java/org/opensearch/index/translog/Translog.java

```diff
@@ -650,6 +650,14 @@ public Snapshot newSnapshot(long fromSeqNo, long toSeqNo, boolean requiredFullRa
                 .filter(reader -> reader.getCheckpoint().minSeqNo <= toSeqNo && fromSeqNo <= reader.getCheckpoint().maxEffectiveSeqNo())
                 .map(BaseTranslogReader::newSnapshot)
                 .toArray(TranslogSnapshot[]::new);
+            if (this instanceof RemoteFsTranslog
+                && fromSeqNo != 0
+                && Arrays.stream(snapshots)
+                    .noneMatch(
+                        snapshot -> snapshot.getCheckpoint().minSeqNo <= fromSeqNo || snapshot.getCheckpoint().maxSeqNo >= fromSeqNo
```

Comment:
1. Let's add a comment to explain this logic.
2. What would happen if the the reader had no operations. Basically we rolled over the generation without any data present in the .tlog file.

Reply:
Lets consider a case where the segment infos has the processed checkpoint as 100. We have the following data in readers - 1. min 90 max 110 2. min 120 max 150. How do we detect that 110-120 is missing.

Reply:
> How do we detect that 110-120 is missing.

It is not possible with this change. This change is only to identify sequence number gap between segment and translog. We should be adding check for the case you mentioned as well.